src/acquisition/flusurv/flusurv.py

```diff
@@ -142,78 +173,192 @@ def mmwrid_to_epiweek(mmwrid):
+        }
+    """
+    data = data["default_data"]
 
-    # iterate over all seasons and age groups
-    for obj in data_in["busdata"]["dataseries"]:
-        if obj["age"] in (10, 11, 12):
-            # TODO(https://github.com/cmu-delphi/delphi-epidata/issues/242):
-            #   capture as-of-yet undefined age groups 10, 11, and 12
-            continue
-        age_index = obj["age"] - 1
-        # iterage over weeks
-        for mmwrid, _, _, rate in obj["data"]:
-            epiweek = mmwrid_to_epiweek(mmwrid)
-            if epiweek not in data_out:
-                # weekly rate of each age group
-                data_out[epiweek] = [None] * 9
-            prev_rate = data_out[epiweek][age_index]
-            if prev_rate is None:
-                # this is the first time to see a rate for this epiweek/age
-                data_out[epiweek][age_index] = rate
-            elif prev_rate != rate:
-                # a different rate was already found for this epiweek/age
-                format_args = (epiweek, obj["age"], prev_rate, rate)
-                print("warning: %d %d %f != %f" % format_args)
-
-    # sanity check the result
-    if len(data_out) == 0:
+    # Sanity check the input. We expect to see some epiweeks
+    if len(data) == 0:
         raise Exception("no data found")
 
-    # print the result and return flu data
-    print(f"found data for {len(data_out)} weeks")
+    id_label_map = make_id_label_map(metadata)
+    id_season_map = make_id_season_map(metadata)
+
+    # Create output object
+    # First layer of keys is epiweeks. Second layer of keys is groups
+    #  (by id, not age in years, sex abbr, etc).
+    #
+    # If a top-level key doesn't already exist, create a new empty dict.
+    # If a secondary key doesn't already exist, create a new key with a
+    #  default value of None if not provided.
+    data_out = defaultdict(lambda: defaultdict(lambda: None))
+
+    # data["default_data"] is a list of dictionaries, with the format
+    #     [
+    #         {'networkid': 1, 'catchmentid': 22, 'seasonid': 49, 'ageid': 0, 'sexid': 0, 'raceid': 1, 'rate': 4.3, 'weeklyrate': 1.7, 'mmwrid': 2493},
+    #         {'networkid': 1, 'catchmentid': 22, 'seasonid': 49, 'ageid': 0, 'sexid': 0, 'raceid': 1, 'rate': 20.3, 'weeklyrate': 0.1, 'mmwrid': 2513},
+    #         {'networkid': 1, 'catchmentid': 22, 'seasonid': 49, 'ageid': 0, 'sexid': 0, 'raceid': 1, 'rate': 20.6, 'weeklyrate': 0.1, 'mmwrid': 2516},
+    #         ...
+    #     ]
+    for obs in data:
+        epiweek = mmwrid_to_epiweek(obs["mmwrid"])
+        season = id_season_map[obs["seasonid"]]
+        groupname = groupids_to_name(
+            ageid = obs["ageid"], sexid = obs["sexid"], raceid = obs["raceid"],
+            id_label_map = id_label_map
+        )
+
+        # Set season description. This will be overwritten every iteration,
+        #  but should always have the same value per epiweek group.
+        data_out[epiweek]["season"] = season
```

Comment:
```suggestion
        data_out[epiweek]["season"] = id_season_map[obs["seasonid"]]
```
and delete the `season = ...` line above (or put this in its place)